### ISSUE TITLE
Bad `.concat` suggestion on `Array`: `+` returns a new array, `.concat` modifies the receiver

### ISSUE THREAD

**Issue by jez** [sorbet/sorbet#6345]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7Bparams%28xs%3A%20T%3A%3AArray%5BInteger%5D%2C%20ys%3A%20T%3A%3AArray%5BString%5D%29.void%7D%0Adef%20example%28xs%2C%20ys%29%0A%20%20xs%20%2B%20ys%0Aend)

```ruby
# typed: true
extend T::Sig

sig {params(xs: T::Array[Integer], ys: T::Array[String]).void}
def example(xs, ys)
  xs + ys
end
```

#### Observed output

```
editor.rb:6: Expected T::Enumerable[Integer] but found T::Array[String] for argument arg0 https://srb.help/7002
     6 |  xs + ys
               ^^
  Expected T::Enumerable[Integer] for argument arg0 of method Array#+:
    https://github.com/sorbet/sorbet/tree/master/rbi/core/array.rbi#L448:
     448 |        arg0: T::Enumerable[Elem],
                  ^^^^
  Got T::Array[String] originating from:
    editor.rb:5:
     5 |def example(xs, ys)
                        ^^
  Note:
    If the desired behavior is to widen the type to T::Array[T.any(Integer, String)], use Array#concat instead
  Autocorrect: Use `-a` to autocorrect
    editor.rb:6: Replace with .concat(ys)
     6 |  xs + ys
            ^^^^^
Errors: 1
```

#### Expected behavior

Sorbet should not subtly change the meaning of a program in an autocorrect.

Either we should

- only fire the autocorrect for `+=` calls, not for normal `+` calls.
- change the suggestion to `[].concat(xs, ys)` instead of `xs + ys` in the non-`+=` case.

- - -

**Comment by maruth-stripe** [comment]
This shape of autocorrect generally seems slightly suspicious to me -- it feels like it makes introducing potential bugs easier. What do you think about just leaving it as an error, as a user might be forced to take another look at whether they really want to merge arrays with distinct types (as opposed to being able to make sorbet "shut up" by running with the autocorrect option)?

**Comment by jez** [comment]
Fixed in #7014
